src/cryptenroll/cryptenroll.c

```diff
@@ -165,6 +168,38 @@ static int help(void) {
         return 0;
 }
 
+static int resolve_auto_node(char **ret) {
+        _cleanup_free_ char *uuid = NULL, *devname = NULL;
+        char sysfs_path[SYS_BLOCK_PATH_MAX("/dm/uuid")];
+        dev_t devnum;
+        int r;
+
+        r = get_block_device("/var", &devnum);
```

Comment:
I don't actually understand this. Why /var/?

Reply:
i think this makes sense. There are primarily three models we try to support with systemd:

1. root read-only, /usr/ on root, /var/ separate and writable
2. root writable, /usr/ separate and read-only, /var/ on root
3. root writable, /usr/ on root, /var/ on root

In all of the above, there's a common theme: /var/ is either itself the only writable partition, or it is part of the root fs, which is writable.

And if encryption is used, then all writable stuff needs to be covered. Hence, an if you want to automatically find the primary encrypted fs we have, then it's a safe bet to just check /var/ as easier this just gets you to the same as rootfs, or if it's split out then it is the right one too look at as rootfs is immutable and thus rather dm-verity'd then dm-crypt'ed...

Reply:
(i mean, people can set up things however they want, but I think for an explicit auto switch it makes sense to reasonably cover the common, realistic, primary scenarios.)

Reply:
/var is the local state partition and thus it covers most cases.

If a distro has a separate encrypted /var partition, then that's what we'll target. In this case, presumably the root partition is either unencrypted or the root partition is a tmpfs (at least those seem the most likely). If /var isn't a separate partition, then it'll neatly resolve to the / partition

This was suggested to me by @poettering

Reply:
OK. Probably worth some comments though.